### PR TITLE
tweaked IPC martial art emp protection

### PR DESCRIPTION
**Pull request by Moltijoe**
sounds like the existing emp mitigation from the martial art wasn't enough to prevent even a single emp from ending your rampage if unlucky

so i've
- given them immunity from the regular emp effect
- added a snowflake check that does damage when emped unless they've adapted
- turns off throw mode if an emp is countered

:cl:  
tweak: IPC martial art now has a snowflake form of emp damage
tweak: IPC martial art turns off throwmode if an emp is countered
/:cl:

### CODE REVIEW

**Comment by KillerOrcaCora** [comment]
Skill issue

**Comment by cowbot92** [comment]
yeah makes sense

**Comment by ReddicusDragon** [comment]
Terrible idea. Powergamey and powercreepy.

**Comment by ReddicusDragon** [comment]
If this genuinely gets merged I will quit yogstation again

**Comment by Moltijoe** [comment]
> If this genuinely gets merged I will quit yogstation again

Perhaps a bit of an overreaction, all this really does is remove the blind and brain damage from getting hit by an emp. Brain damage is such a dice role and annoying to heal that it can effectively be an instant kill on the ipc.

Also, if the martial art ends up too strong, nerfs CAN be made. No clue why people treat every PR like they are final and permanent.

edit: also its a traitor martial art, those are innately powergamey.

**Comment by ReddicusDragon** [comment]
I can guarantee to you it will be too strong. You can already deflect EMPs using throw mode so why the fuck do you need to be EMP prove.

**Comment by Moltijoe** [comment]
> I can guarantee to you it will be too strong. You can already deflect EMPs using throw mode so why the fuck do you need to be EMP prove.

Its not emp proof, it just removes the effects other than the damage. Also, tell me, when was the last time you were able to react to someone hand mixing iron and uranium?

**Comment by Marmio64** [comment]
> > I can guarantee to you it will be too strong. You can already deflect EMPs using throw mode so why the fuck do you need to be EMP prove.
> 
> Its not emp proof, it just removes the effects other than the damage. Also, tell me, when was the last time you were able to react to someone hand mixing iron and uranium?

reacted to it by playing human

**Comment by Moltijoe** [comment]
> reacted to it by playing human

bruh

**Comment by ReddicusDragon** [comment]
mmm i should test this sometime

**Comment by MajManatee** [comment]
air is getting awfully stale around here...

**Comment by ReddicusDragon** [comment]
> air is getting awfully stale around here...

sounds like you should review it

**Comment by ReddicusDragon** [comment]
@Moltijoe fix conflicts

**Comment by Moltijoe** [comment]
> @Moltijoe fix conflicts

bruh what? literally nothing changed for there to be a conflict
must have been one of those end of file line things

File: code/modules/mob/living/carbon/human/species_types/IPC.dm [hunk 1]
```diff
@@ -266,27 +269,39 @@ ipc martial arts stuff
...
-/datum/species/ipc/proc/add_empproof(mob/living/carbon/human/H)
-	H.AddComponent(/datum/component/empprotection, EMP_PROTECT_SELF | EMP_PROTECT_CONTENTS)
-
-/datum/species/ipc/proc/remove_empproof(mob/living/carbon/human/H)
-	var/datum/component/empprotection/ipcmartial = H.GetExactComponent(/datum/component/empprotection)
-	if(ipcmartial)
-		ipcmartial.Destroy()
-
-#undef CONSCIOUSAY
+#undef CONCIOUSAY
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/IPC.dm]:
```suggestion
#undef CONSCIOUSAY
```

**Comment by Moltijoe** [comment]
might reopen this if i feel it's still needed

**Pull request closed without merging**